lib/fluent/plugin/out_http.rb

```diff
@@ -323,7 +328,26 @@ def create_request(chunk, uri)
               Net::HTTP::Put.new(uri.request_uri)
             end
       set_headers(req, uri, chunk)
-      req.body = @json_array ? "[#{chunk.read.chop}]" : chunk.read
+
+      compress_body = @compress != :text
+
+      case 
+      when @compress != @buffer.compress
+        req.body = @json_array ? "[#{chunk.read.chop}]" : chunk.read
+      when @compress == @buffer.compress
+        if @json_array
+          req.body = "[#{chunk.read.chop}]"
+        else
+          req.body = chunk.read(compressed: @compress)
+          compress_body = false
+        end
+      end
+
+      if compress_body
+        gz = Zlib::GzipWriter.new(StringIO.new)
+        gz << req.body
+        req.body = gz.close.string
+      end
```

Comment:
Ok, so the json_array setting throws a bit of a spanner in the works, but the gist here is that if both out_http and buffer are set to gzip and json_array isnt set, then send the gzipped chunk as is.

Otherwise decompress (if required) and recompress (if required).

Reply:
Thanks!
I understand the specification!

Reply:
Sorry @daipom, I just updated this section to tidy it up a little more and make it easier to read

Reply:
I see! No problem!